<launch>
	<node name="get_map_server" pkg="autonomous_exploration" type="get_map_server" />

	<!-- Robot's radius (meters). -->
	<param name="robot_radius" type="double" value="$(optenv ROS_ROBOT_RADIUS 0.18)" />

	<!-- Laser's max range (meters). -->
	<param name="laser_max_range" type="double" value="$(optenv ROS_LASER_MAX_RANGE 3.0)" />

	<!-- How far the point needed to be to consider it as a frontier (meters). -->
	<param name="min_distance" type="double" value="$(optenv ROS_MIN_DISTANCE 1.0)" />

	<!-- Obstacle threshold, cell's value below lethal_cost is considered as free, otherwise occupied (0..100). -->
	<param name="lethal_cost" type="int" value="$(optenv ROS_LETHAL_COST 65)" />

	<!-- Default gain threshold to consider the point as frontier (%). It dynamically decreases or increases by factor of gain_change factor. -->
	<param name="gain_threshold" type="double" value="$(optenv ROS_GAIN_THRESHOLD 35.0)" />
	<param name="gain_change" type="double" value="$(optenv ROS_GAIN_CHANGE 1.5)" />

	<!-- The minimum value of gain_threshold can decrease down to (%). -->
	<param name="min_gain_threshold" type="double" value="$(optenv ROS_MIN_GAIN_THRESHOLD 10.0)" />

	<!-- The directory in which the map will be saved at the end of exploration. -->
	<param name="map_path" type="string" value="$(env HOME)/maps/"/>

	<node name="explore_server" pkg="autonomous_exploration" type="explore_server" output="screen" />
</launch>
